<launch>

	<!--  Pepper Launch  -->
	<include file="$(find printeps_pepper_modules)/launch/pepper_setup.launch">
		<arg name="nao_ip"              value="$(optenv NAO_IP 192.168.1.16)" />
		<!--arg name="roscore_ip"          value="192.168.1.2" /-->
		<arg name="network_interface"   value="eth0" />
	</include>


  <!-- Robot LRS Offset Broadcaster -->
	<node pkg="printeps_pepper_modules" type="pepper_tf_broadcaster" name="pepper_tf_broadcaster"/>


  <!-- Laser Range launch  -->
  <node ns="lrs_front" pkg="urg_node" type="urg_node" name="lrs0">
    <param name="ip_address" value="192.168.1.17"/>
    <param name="frame_id" 	value="base_scan"/>
  </node>


  <!-- provide map -->
  <node name="map_server" pkg="map_server" type="map_server"
      args="$(find printeps_pepper_modules)/map/map.yaml" respawn="false" >
      <param name="use_map_topic"   value="true"/>
      <param name="frame_id" 				value="map" />
  </node>


  <!-- localizaion -->
  <node pkg="amcl" type="amcl" name="amcl" output="screen">
      <remap from="scan" to="lrs_front/scan" />
			<param name="odom_frame_id"		value="odom"/>
			<param name="base_frame_id"		value="base_footprint"/>
			<param name="global_frame_id"	value="map"/>
      <rosparam file="$(find printeps_pepper_modules)/config/amcl.yaml" command="load" />
  </node>


	<!-- path planning -->
  <include file="$(find fetch_navigation)/launch/include/move_base.launch.xml" >
      <arg name="name" 					value="fetch" />
			<arg name="map_topic" 		value="map" />
			<arg name="odom_topic" 		value="odom" />
			<arg name="cmd_vel_topic" value="cmd_vel" />
  </include>
	<rosparam>
    move_base/TrajectoryPlannerROS/max_vel_x: 0.25
    move_base/TrajectoryPlannerROS/min_vel_x: 0.08
    move_base/TrajectoryPlannerROS/max_vel_theta: 0.15
    move_base/TrajectoryPlannerROS/min_vel_theta: -0.15
    move_base/TrajectoryPlannerROS/recovery_behavior_enabled: false
    move_base/local_costmap/obstacles/base_scan/max_obstacle_height: 0.4
    move_base/local_costmap/obstacles/base_scan/min_obstacle_height: -0.2
    move_base/local_costmap/inflater/inflation_radius: 0.5
    move_base/global_costmap/inflater/inflation_radius: 0.5
  </rosparam>


  <!-- Joystick motion -->
  <arg name="joy_dev" default="/dev/input/js0" />

  <node pkg="joy" type="joy_node" name="joy_node">
    <param name="dev" value="$(arg joy_dev)" />
    <param name="autorepeat_rate" value="5" />
  </node>

  <node pkg="printeps_pepper_modules" name="teleop_twist_joy" type="teleop_twist_joy.py" >
		<param name="max_speed"	value="0.25" />
		<param name="max_turn"	value="0.8" />
	</node>



	<!-- Pepper Navigatoin service -->
  <node pkg="printeps_pepper_modules" name="pepper_navigation" type="pepper_navigation.py" />



  <!-- Display -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find printeps_pepper_modules)/rviz/pepper_motion.rviz" />


</launch>
